cardano-api/src/Cardano/Api/Eras.hs

```diff
@@ -365,6 +383,62 @@ instance IsShelleyBasedEra BabbageEra where
 instance IsShelleyBasedEra ConwayEra where
    shelleyBasedEra = ShelleyBasedEraConway
 
+data AnyShelleyBasedEra where
```

Comment:
Why GADT instead of newtype?

Reply:
How would you encode this as a `newtype`?